### PR TITLE
Add housekeeping configuration of jobs containing errors

### PR DESCRIPTION
**Pull request by fviale**
Also, improved adjusting the housekeeping after job recovery (when enabled)

### CODE REVIEW

**Review by mboussaa — APPROVED** [review]
LGTM

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java [hunk 1]
```diff
@@ -812,6 +812,21 @@ private void setJobStatusAfterInErrorTaskRestart(InternalJob job) {
         }
     }
 
+    public static boolean isJobWithErrors(InternalJob job) {
+        switch (job.getStatus()) {
+            case FAILED:
+            case CANCELED:
+                return true;
+            default:
+                if (job.getJobInfo().getNumberOfFailedTasks() > 0 || job.getJobInfo().getNumberOfFaultyTasks() > 0) {
```
Comment (tao-xinxiu) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java:821]: The whole if-else part could be simplified to `return job.getJobInfo().getNumberOfFailedTasks() > 0 || job.getJobInfo().getNumberOfFaultyTasks() > 0`

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java [hunk 2]
```diff
@@ -1128,6 +1129,29 @@ void terminateJobHandling(final JobId jobId, final Map<String, String> jobGeneri
         }
     }
 
+    private long getRemoveDelayFromGenericInfo(JobId jobId, Map<String, String> jobGenericInfo, boolean isJobWithErrors,
+            long removeDelay) {
```
Comment (tao-xinxiu) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java:1133]: This function seems not need the variable `removeDelay` as input?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java [hunk 3]
```diff
@@ -1128,6 +1129,29 @@ void terminateJobHandling(final JobId jobId, final Map<String, String> jobGeneri
...
+        if (jobGenericInfo != null) {
+            if (jobGenericInfo.containsKey(GENERIC_INFO_REMOVE_DELAY)) {
+                try {
+                    removeDelay = Tools.parsePeriod(jobGenericInfo.get(GENERIC_INFO_REMOVE_DELAY));
+                } catch (Exception e) {
+                    logger.error("Error when parsing generic information " + GENERIC_INFO_REMOVE_DELAY + " for job " +
+                                 jobId, e);
+                }
+            }
+            if (isJobWithErrors && jobGenericInfo.containsKey(GENERIC_INFO_REMOVE_DELAY_ON_ERROR)) {
```
Comment (tao-xinxiu) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java:1143]: Could the two cases both happen? `else if` might be a bit more clear?

**Review by tao-xinxiu — APPROVED** [review]

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java (hunk 1 above)
Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java:821]: indeed

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java (hunk 3 above)
Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java:1143]: yes both case can happen. A user can specific both generic info

**Pull request merged**